packages/pglite/src/pglite.ts

```diff
@@ -600,13 +600,13 @@ export class PGlite
     // >0 set buffer content type to wire protocol
     mod._use_wire(1)
     // set buffer size so answer will be at size+0x2 pointer addr
-    mod._interactive_write(msg_len)
+    // mod._interactive_write(msg_len)
 
     // copy whole buffer at addr 0x1
     mod.HEAPU8.set(message, 1)
 
     // execute the message
-    mod._interactive_one()
+    mod._interactive_one(msg_len)
```

Comment:
my 2 cents: if possible try to pass all parameters in a prepare function like maybe use_wire() before calling the hot loop. The less c-stack that one has ; the better it will be if more unrolling is needed in future.